### ISSUE TITLE
[FP] Suspicious assignments: variable immediately overriden

### ISSUE THREAD

**Issue by orklah** [kalessil/phpinspectionsea#1225]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate), 2.0.13 |

Current behaviour
--
Hi! Here's a weird case

```php
<?php

function foo($bar = null)
{
    if (rand()) {
        $bar = true;
        if(rand()){
            return $bar;
        }
        else{
            return null;
        }
    }
    $bar = false;
    return $bar;
}
```
$bar = false is detected as an overriding of $bar = true

Thanks!

**Comment by kalessil** [comment]
Taking into account that the preceding if-statement is optional, the inspection behaves as expected. Hence closing the issues.

**Comment by orklah** [comment]
Thanks for looking into it! However, I'm not sure I understand what you mean. How do you suggest to change the code to fix the issue?
